src/xrc/src/utils.rs

```diff
@@ -59,16 +59,16 @@ pub(crate) fn standard_deviation(rates: &[u64]) -> u64 {
     if count < 2 {
         return 0;
     }
-    let mean: i64 = (rates
+    let mean = (rates
         .iter()
         .map(|rate| *rate as u128)
         .sum::<u128>()
-        .saturating_div(count as u128)) as i64;
+        .saturating_div(count as u128)) as i128;
     let variance = rates
         .iter()
-        .map(|rate| (((*rate as i64).saturating_sub(mean)).pow(2)) as u64)
-        .sum::<u64>()
-        .saturating_div(count.saturating_sub(1));
+        .map(|rate| ((*rate as i128).saturating_sub(mean)).saturating_pow(2) as u128)
+        .sum::<u128>()
+        .saturating_div(count.saturating_sub(1) as u128);
     (variance as f64).sqrt() as u64
```

Comment:
Is this the only place where we use a floating-point operation? Parsing and displaying floating-point numbers don't count.   :slightly_smiling_face:

I think we can get rid of this operation here. I'll try to address it in a separate PR.

Reply:
This is the only place if we are not counting the multiplying that occurs when extracting rates.